### DISCUSSION TITLE
Is it okay to add arbitrary methods on pydantic model class

### DISCUSSION THREAD

**Discussion by saumzzz** [pydantic/pydantic#9005] [Question] [answered]
In v1, as answered here(https://stackoverflow.com/a/60048805) by @samuelcolvin, it was alright to add arbitrary methods to a pydantic model class
Is that still the case or is there a more pydantic-v2 way of doing things or considered bad practice altogether?

I searched for this in issues and discussions, didn't find the answer, sorry if it's been answered before

**Comment by Viicos** [comment, marked as answer]
It should be completely fine in v2 as well, you should just try to avoid using the protected `model_*` namespace, e.g. avoid defining `model_my_custom_method(...)`
